api/envoy/config/core/v3/address.proto

```diff
@@ -98,9 +98,18 @@ message SocketAddress {
   // IPv6 space as ``::FFFF:<IPv4-address>``.
   bool ipv4_compat = 6;
 
-  // The Linux network namespace to bind the socket to. If this is set, Envoy will
-  // create the socket in the specified network namespace. Only supported on Linux.
-  // [#not-implemented-hide:]
+  // Filepath that specifies the Linux network namespace this socket will be created in (see ``man 7
+  // network_namespaces``).  If this field is set, Envoy will create the socket in the specified
+  // network namespace.
+  //
+  // .. note::
+  //    Setting this parameter requires Envoy to run with the ``CAP_NET_ADMIN`` capability.
+  //
+  // .. note::
+  //    Currently only used for Listener sockets.
+  //
+  // .. attention::
+  //     Network namespaces are only configurable on Linux. Otherwise, this field has no effect.
```

Comment:
Should the client NACK if this is set but it can't support it (either because it's not running on Linux or because it's not running with the necessary capability)?

Reply:
AFAIK there's also a similar case with the `transparent` field in the listener. If you set it without proper capabilities envoy will just crash with a permission denied error.